.github/workflows/build-storybook-main.yml

```diff
@@ -0,0 +1,45 @@
+name: Build Storybook Main
+
+on:
+  push:
+    branches:
+      - main
+  workflow_call:
+    secrets:
+      PUBLISH_STORYBOOK_TOKEN:
+        required: true
+
+jobs:
+  build-deploy-storybook:
+    name: Build and Deploy Storybook
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    environment: github-pages
+    defaults:
+      run:
+        working-directory: apps/storybook-react
+    steps:
+      - name: Checkout and setup environment
+        uses: MetaMask/action-checkout-and-setup@v1
+        with:
+          is-high-risk-environment: false
+          cache-node-modules: true
```

Comment:
Not sure if there is an alternative here to using `cache-node-modules` I've seen in slack discussions that this might be unsafe. Maybe @itsyoboieltr knows?

Reply:
Thanks for the detailed explanation and suggestion. I've updated here https://github.com/MetaMask/metamask-design-system/pull/610/commits/024a16f1a502a02ff7a93bc9f3f51c93beb6631d